### ISSUE TITLE
cant get data

### ISSUE THREAD

**Issue by codeguru1983** [5ymph0en1x/Heptet#3]
Hi,
First of all congratulations for this project, brilliant idea.
I've just finished installing everything and I get these errors, I think a step by step which file should be execute will make it a lot easier for everybody. Anyway this is what I get:
```

Wrote to ./DAT_ASCII_USDCAD_M1_201908.zip
https://www.histdata.com/download-free-forex-historical-data/?/ascii/1-minute-bar-quotes/usdcad/2019/9
{'tk': 'd477a807c601da1827f549e96ecb6c74', 'date': '2019', 'datemonth': '201909', 'platform': 'ASCII', 'timeframe': 'M1', 'fxpair': 'USDCAD'}
Wrote to ./DAT_ASCII_USDCAD_M1_201909.zip
https://www.histdata.com/download-free-forex-historical-data/?/ascii/1-minute-bar-quotes/usdcad/2019/10
{'tk': 'c769445b0c6f2d8bdc61673b2ecdef61', 'date': '2019', 'datemonth': '201910', 'platform': 'ASCII', 'timeframe': 'M1', 'fxpair': 'USDCAD'}
Wrote to ./DAT_ASCII_USDCAD_M1_201910.zip
Traceback (most recent call last):
  File "formater_piphist.py", line 69, in <module>
    ts = ts.reindex(index=index, method='pad')
  File "/usr/lib/python3/dist-packages/pandas/util/_decorators.py", line 127, in wrapper
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/pandas/core/frame.py", line 2935, in reindex
    return super(DataFrame, self).reindex(**kwargs)
  File "/usr/lib/python3/dist-packages/pandas/core/generic.py", line 3023, in reindex
    fill_value, copy).__finalize__(self)
  File "/usr/lib/python3/dist-packages/pandas/core/frame.py", line 2870, in _reindex_axes
    fill_value, limit, tolerance)
  File "/usr/lib/python3/dist-packages/pandas/core/frame.py", line 2878, in _reindex_index
    tolerance=tolerance)
  File "/usr/lib/python3/dist-packages/pandas/core/indexes/base.py", line 2993, in reindex
    raise ValueError("cannot reindex a non-unique index "
ValueError: cannot reindex a non-unique index with a method or limit
```
I do have however some files saved now csv's inside the Symphoenix folder.
Thank you

**Comment by 5ymph0en1x** [comment]
Hello !
Thanks for your comment.
Try moving the scripts from the Symphoenix folder to the root one of this project before running them.
Best regards

**Comment by codeguru1983** [comment]
Hi,
Sorry to be a pain but it seems that I can't get it right. I've reformatted and installed a fresh copy of Ubuntu 18.04, I've created my own virtual environment, and then used your requirements.txt

Unfortunatelly I think it installs directly tensorflow 2 so I had to downgrade that to 1.5, I'm not sure for each of these requirements which versions you used.

After moving the files in the root of the project I still get the same errors, however, I've downloaded the CSVs by hand, merged them into 1 file for 2019 for each of those 2 pairs.

Can you please offer me a few minutes of your time and help me to understand which versions you are using?
```

virtual environment
python
pip
----------
pandas
numpy
scipy
scikit-learn
matplotlib
tensorflow
aiohttp
statsmodels
PyYAML
seaborn
oandapyV20
pyTelegramBotAPI
python-dateutil
histdata
```

With the 1.5 tensorflow I get these errors, the 2.0 will just give that core error.

```
(myenv) catalin@dell:~/Documents/heptet$ python Training_Model.py
/home/catalin/myenv/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:493: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/home/catalin/myenv/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:494: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/home/catalin/myenv/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:495: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/home/catalin/myenv/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:496: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint16 = np.dtype([("quint16", np.uint16, 1)])
/home/catalin/myenv/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:497: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint32 = np.dtype([("qint32", np.int32, 1)])
/home/catalin/myenv/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:502: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  np_resource = np.dtype([("resource", np.ubyte, 1)])
Traceback (most recent call last):
  File "Training_Model.py", line 12, in <module>
    from STRATEGY.Cointegration import EGCointegration
  File "/home/catalin/Documents/heptet/STRATEGY/Cointegration.py", line 35, in <module>
    config_trading = FileIO.read_yaml(config_path)
  File "/home/catalin/Documents/heptet/UTIL/FileIO.py", line 19, in read_yaml
    with open(file_path, 'r') as stream:
FileNotFoundError: [Errno 2] No such file or directory: 'CONFIG\\config_trading.yml'

```
Thank you very much, I promise that I'll create a tutorial on how to install it from scratch on a fresh copy of Ubuntu 18.04 once I get my head around.

**Comment by codeguru1983** [comment]
This is what I have in 
$ pip list

```
(myenv) catalin@dell:~/Documents/heptet$ pip list
Package                Version   
---------------------- ----------
absl-py                0.8.1     
aiohttp                3.6.2     
astor                  0.8.1     
async-timeout          3.0.1     
attrs                  19.3.0    
beautifulsoup4         4.8.1     
bleach                 1.5.0     
cachetools             3.1.1     
certifi                2019.11.28
chardet                3.0.4     
cycler                 0.10.0    
gast                   0.2.2     
google-auth            1.8.2     
google-auth-oauthlib   0.4.1     
google-pasta           0.1.8     
grpcio                 1.25.0    
h5py                   2.10.0    
histdata               1.0       
html5lib               0.9999999 
idna                   2.8       
idna-ssl               1.1.0     
joblib                 0.14.1    
Keras-Applications     1.0.8     
Keras-Preprocessing    1.1.0     
kiwisolver             1.1.0     
Markdown               3.1.1     
matplotlib             3.1.2     
multidict              4.7.1     
numpy                  1.17.4    
oandapyV20             0.6.3     
oauthlib               3.1.0     
opt-einsum             3.1.0     
pandas                 0.25.3    
patsy                  0.5.1     
pip                    19.3.1    
pkg-resources          0.0.0     
protobuf               3.11.1    
pyasn1                 0.4.8     
pyasn1-modules         0.2.7     
pyparsing              2.4.5     
pyTelegramBotAPI       3.6.6     
python-dateutil        2.8.1     
pytz                   2019.3    
PyYAML                 5.2       
requests               2.22.0    
requests-oauthlib      1.3.0     
rsa                    4.0       
scikit-learn           0.22      
scipy                  1.3.3     
seaborn                0.9.0     
setuptools             42.0.2    
six                    1.13.0    
soupsieve              1.9.5     
statsmodels            0.10.2    
tensorboard            2.0.2     
tensorflow             1.5.0     
tensorflow-estimator   2.0.1     
tensorflow-tensorboard 1.5.1     
termcolor              1.1.0     
typing-extensions      3.7.4.1   
urllib3                1.25.7    
Werkzeug               0.16.0    
wheel                  0.33.6    
wrapt                  1.11.2    
yarl                   1.4.2  
```

**Comment by codeguru1983** [comment]
I've also tried to upgrade to tensorflow 2.0 with: 
`(myenv) catalin@dell:~/Documents/heptet$ pip install --upgrade tensorflow`
after installation my pip list looks like:
```
(myenv) catalin@dell:~/Documents/heptet$ pip list
Package                Version   
---------------------- ----------
absl-py                0.8.1     
aiohttp                3.6.2     
astor                  0.8.1     
async-timeout          3.0.1     
attrs                  19.3.0    
beautifulsoup4         4.8.1     
bleach                 1.5.0     
cachetools             3.1.1     
certifi                2019.11.28
chardet                3.0.4     
cycler                 0.10.0    
gast                   0.2.2     
google-auth            1.8.2     
google-auth-oauthlib   0.4.1     
google-pasta           0.1.8     
grpcio                 1.25.0    
h5py                   2.10.0    
histdata               1.0       
html5lib               0.9999999 
idna                   2.8       
idna-ssl               1.1.0     
joblib                 0.14.1    
Keras-Applications     1.0.8     
Keras-Preprocessing    1.1.0     
kiwisolver             1.1.0     
Markdown               3.1.1     
matplotlib             3.1.2     
multidict              4.7.1     
numpy                  1.17.4    
oandapyV20             0.6.3     
oauthlib               3.1.0     
opt-einsum             3.1.0     
pandas                 0.25.3    
patsy                  0.5.1     
pip                    19.3.1    
pkg-resources          0.0.0     
protobuf               3.11.1    
pyasn1                 0.4.8     
pyasn1-modules         0.2.7     
pyparsing              2.4.5     
pyTelegramBotAPI       3.6.6     
python-dateutil        2.8.1     
pytz                   2019.3    
PyYAML                 5.2       
requests               2.22.0    
requests-oauthlib      1.3.0     
rsa                    4.0       
scikit-learn           0.22      
scipy                  1.3.3     
seaborn                0.9.0     
setuptools             42.0.2    
six                    1.13.0    
soupsieve              1.9.5     
statsmodels            0.10.2    
tensorboard            2.0.2     
tensorflow             2.0.0     
tensorflow-estimator   2.0.1     
tensorflow-tensorboard 1.5.1     
termcolor              1.1.0     
typing-extensions      3.7.4.1   
urllib3                1.25.7    
Werkzeug               0.16.0    
wheel                  0.33.6    
wrapt                  1.11.2    
yarl                   1.4.2    
```

but trying to use $ python Training_Model.py will return:
`Illegal instruction (core dumped)`

After this I've tried to update the -tensorboard to 2, but same error occurs. 
Thank you again!

**Comment by codeguru1983** [comment]
this is the output for cat /proc/cpuinfo
```

processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 55
model name	: Intel(R) Celeron(R) CPU  N2840  @ 2.16GHz
stepping	: 8
microcode	: 0x838
cpu MHz		: 1658.286
cache size	: 1024 KB
physical id	: 0
siblings	: 2
core id		: 0
cpu cores	: 2
apicid		: 0
initial apicid	: 0
fpu		: yes
fpu_exception	: yes
cpuid level	: 11
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology tsc_reliable nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm sse4_1 sse4_2 movbe popcnt tsc_deadline_timer rdrand lahf_lm 3dnowprefetch epb pti ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid tsc_adjust smep erms dtherm ida arat md_clear
bugs		: cpu_meltdown spectre_v1 spectre_v2 mds msbds_only
bogomips	: 4331.60
clflush size	: 64
cache_alignment	: 64
address sizes	: 36 bits physical, 48 bits virtual
power management:

processor	: 1
vendor_id	: GenuineIntel
cpu family	: 6
model		: 55
model name	: Intel(R) Celeron(R) CPU  N2840  @ 2.16GHz
stepping	: 8
microcode	: 0x838
cpu MHz		: 1693.612
cache size	: 1024 KB
physical id	: 0
siblings	: 2
core id		: 1
cpu cores	: 2
apicid		: 2
initial apicid	: 2
fpu		: yes
fpu_exception	: yes
cpuid level	: 11
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology tsc_reliable nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm sse4_1 sse4_2 movbe popcnt tsc_deadline_timer rdrand lahf_lm 3dnowprefetch epb pti ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid tsc_adjust smep erms dtherm ida arat md_clear
bugs		: cpu_meltdown spectre_v1 spectre_v2 mds msbds_only
bogomips	: 4331.60
clflush size	: 64
cache_alignment	: 64
address sizes	: 36 bits physical, 48 bits virtual
power management:
```

**Comment by 5ymph0en1x** [comment]
I hope this isn't a platform dependant issue...
Here's a list of the versions :

```
Package                            Version

---------------------------------- ---------
absl-py                            0.8.1
aiohttp                            3.6.2
alabaster                          0.7.12
anaconda-client                    1.7.2
anaconda-project                   0.8.3
asn1crypto                         1.0.1
astor                              0.8.0
astroid                            2.3.1
astropy                            3.2.2
async-timeout                      3.0.1
atomicwrites                       1.3.0
attrs                              19.2.0
audioread                          2.1.8
autobahn                           19.10.1
Automat                            0.8.0
Babel                              2.7.0
backcall                           0.1.0
backports.os                       0.1.1
backports.shutil-get-terminal-size 1.0.0
backports.weakref                  1.0rc1
beautifulsoup4                     4.8.0
bitarray                           1.0.1
bitfinex-v2                        1.6.0
bkcharts                           0.2
bleach                             1.5.0
bokeh                              1.3.4
boto                               2.49.0
Bottleneck                         1.2.1
certifi                            2019.9.11
cffi                               1.12.3
chardet                            3.0.4
Click                              7.0
cloudpickle                        1.2.2
clyent                             1.2.2
colorama                           0.4.1
comtypes                           1.1.7
constantly                         15.1.0
contextlib2                        0.6.0
cryptography                       2.7
cycler                             0.10.0
Cython                             0.29.13
cytoolz                            0.10.0
dask                               2.6.0
decorator                          4.4.0
defusedxml                         0.6.0
distributed                        2.6.0
docutils                           0.15.2
entrypoints                        0.3
enum34                             1.1.6
et-xmlfile                         1.0.1
fastcache                          1.1.0
Flask                              1.1.1
fsspec                             0.5.2
gast                               0.2.2
gevent                             1.4.0
google-pasta                       0.1.7
greenlet                           0.4.15
grpcio                             1.24.1
h5py                               2.9.0
HeapDict                           1.0.1
histdata                           1.0
html5lib                           0.9999999
httpretty                          0.9.7
hyperlink                          19.0.0
idna                               2.8
idna-ssl                           1.1.0
imageio                            2.6.0
imagesize                          1.1.0
importlib-metadata                 0.23
incremental                        17.5.0
ipykernel                          5.1.2
ipython                            7.8.0
ipython-genutils                   0.2.0
ipywidgets                         7.5.1
isort                              4.3.21
itsdangerous                       1.1.0
jdcal                              1.4.1
jedi                               0.15.1
Jinja2                             2.10.3
joblib                             0.13.2
json5                              0.8.5
jsonschema                         3.0.2
jupyter                            1.0.0
jupyter-client                     5.3.4
jupyter-console                    6.0.0
jupyter-core                       4.6.0
jupyterlab                         1.1.4
jupyterlab-server                  1.0.6
Keras                              2.1.4
Keras-Applications                 1.0.8
Keras-Preprocessing                1.1.0
keyring                            18.0.0
kiwisolver                         1.1.0
lazy-object-proxy                  1.4.2
librosa                            0.7.1
llvmlite                           0.29.0
locket                             0.2.0
Logbook                            1.5.3
lxml                               4.4.1
Markdown                           3.1.1
MarkupSafe                         1.1.1
matplotlib                         3.1.1
mccabe                             0.6.1
menuinst                           1.4.16
mistune                            0.8.4
mkl-fft                            1.0.14
mkl-random                         1.1.0
mkl-service                        2.3.0
mock                               3.0.5
more-itertools                     7.2.0
mpmath                             1.1.0
msgpack                            0.6.1
multidict                          4.5.2
multipledispatch                   0.6.0
nbconvert                          5.6.0
nbformat                           4.4.0
networkx                           2.3
nltk                               3.4.5
nose                               1.3.7
notebook                           6.0.1
numba                              0.45.1
numexpr                            2.7.0
numpy                              1.16.5
numpydoc                           0.9.1
oandapyV20                         0.6.3
olefile                            0.46
openpyxl                           3.0.0
opt-einsum                         3.1.0
packaging                          19.2
pandas                             0.22.0
pandas-datareader                  0.8.1
pandocfilters                      1.4.2
parso                              0.5.1
partd                              1.0.0
path.py                            12.0.1
pathlib2                           2.3.5
patsy                              0.5.1
pep8                               1.7.1
pickleshare                        0.7.5
Pillow                             6.2.0
pip                                19.2.3
pluggy                             0.13.0
ply                                3.11
prometheus-client                  0.7.1
prompt-toolkit                     2.0.10
protobuf                           3.10.0
psutil                             5.6.3
py                                 1.8.0
pyasn1                             0.4.7
pyasn1-modules                     0.2.7
pycodestyle                        2.5.0
pycosat                            0.6.3
pycparser                          2.19
pycrypto                           2.6.1
pycurl                             7.43.0.3
pyflakes                           2.1.1
Pygments                           2.4.2
PyHamcrest                         1.9.0
pylint                             2.4.2
pyodbc                             4.0.27
pyOpenSSL                          19.0.0
pyparsing                          2.4.2
pyreadline                         2.1
pyrsistent                         0.15.4
PySocks                            1.7.1
pyTelegramBotAPI                   3.6.6
pytest                             5.2.1
pytest-arraydiff                   0.3
pytest-astropy                     0.5.0
pytest-doctestplus                 0.4.0
pytest-openfiles                   0.4.0
pytest-remotedata                  0.3.2
python-dateutil                    2.8.0
python-speech-features             0.6
pytz                               2019.3
PyWavelets                         1.0.3
pywin32                            223
pywinpty                           0.5.5
PyYAML                             5.1.2
pyzmq                              18.1.0
QtAwesome                          0.6.0
qtconsole                          4.5.5
QtPy                               1.9.0
requests                           2.22.0
requests-file                      1.4.3
resampy                            0.2.2
rope                               0.14.0
ruamel-yaml                        0.15.46
scikit-image                       0.15.0
scikit-learn                       0.21.3
scipy                              1.3.1
seaborn                            0.9.0
Send2Trash                         1.5.0
service-identity                   18.1.0
setuptools                         41.4.0
simplegeneric                      0.8.1
singledispatch                     3.4.0.3
six                                1.12.0
snowballstemmer                    2.0.0
sortedcollections                  1.1.2
sortedcontainers                   2.1.0
SoundFile                          0.10.2
soupsieve                          1.9.3
Sphinx                             2.2.0
sphinxcontrib-applehelp            1.0.1
sphinxcontrib-devhelp              1.0.1
sphinxcontrib-htmlhelp             1.0.2
sphinxcontrib-jsmath               1.0.1
sphinxcontrib-qthelp               1.0.2
sphinxcontrib-serializinghtml      1.1.3
sphinxcontrib-websupport           1.1.2
spyder                             3.3.6
spyder-kernels                     0.5.2
SQLAlchemy                         1.3.9
statsmodels                        0.10.1
sympy                              1.4
tables                             3.5.2
tblib                              1.4.0
tensorboard                        1.14.0
tensorflow-estimator               1.14.0rc0
tensorflow-gpu                     1.4.0
tensorflow-tensorboard             0.4.0
termcolor                          1.1.0
terminado                          0.8.2
testpath                           0.4.2
toolz                              0.10.0
tornado                            6.0.3
traitlets                          4.3.3
Twisted                            19.7.0
txaio                              18.8.1
typed-ast                          1.4.0
typing-extensions                  3.7.4
unicodecsv                         0.14.1
urllib3                            1.24.2
wcwidth                            0.1.7
webencodings                       0.5.1
Werkzeug                           0.16.0
wheel                              0.33.6
widgetsnbextension                 3.5.1
win-inet-pton                      1.1.0
win-unicode-console                0.5
wincertstore                       0.2
wrapt                              1.11.2
xlrd                               1.2.0
XlsxWriter                         1.2.1
xlwings                            0.15.10
xlwt                               1.3.0
yarl                               1.3.0
zict                               1.0.0
zipp                               0.6.0
zope.interface                     4.6.0
```

**Comment by codeguru1983** [comment]
Hi,
I think it has to do with tenserflow GPU / CPU ... I have CPU not GPU on this laptop I don't have an nvidia video card, do you know if I can sort it from code?

**Comment by 5ymph0en1x** [comment]
Look at the error message in your second post : 

`FileNotFoundError: [Errno 2] No such file or directory: 'CONFIG\\config_trading.yml'`

This isn't related to TF so there is hope :-)
Maybe a path convention issue...

**Comment by 5ymph0en1x** [comment]
By the way, I just fixed the issue with `formater_piphist.py`.

**Comment by codeguru1983** [comment]
Thank you very much sir.

I managed to get it working in the end with a dozens of warnings when
running the training, right now I'm running the trading one.
I'm not sure if I'm on the right path I have a message with It began in
Africa ... then something about my CPU which supports instructions that
this TensorFlow binary was not compiled to use: SSE 4.1 SSE 4.2 AVX AVX2
and so on... then minute update... ready to analyze history and it has
EUR_JPY: C1199, TS: 2019-12-15 22:27 / TE 2019 12 16 19:39 and so on so I
suppose I'm nearly there but on most of them it sais no trading opportunity
spotted. p_value needs to be less than 0.1...

anyway, I'll wait to see what happens in the end and what steps I need to
take further to see that BUY / SELL on the screen eventually.

In the end I took a share VPS server and it worked, I did everything in
small steps and I've got here, I'll see how it goes and keep you posted.

Not sure what's next .

Kind regards,
Cata
